gsr_booking/booking_logic.py

```diff
@@ -0,0 +1,255 @@
+import requests
+import datetime
+import math
+import random
+
+
+class BookingLogic:
+
+    DATE_FORMAT_STR = "%Y-%m-%dT%H:%M:%S%z"
+    MAX_SLOT_HRS = 2.0  # the longest booking allowed per person
+    MIN_SLOT_HRS = 0.5  # the minimum booking allowed per person
+
+    def book_room_for_group(self, group, is_wharton, room, lid, start, end, requester_pennkey):
+        # makes a request to labs api server to book rooms, and returns result json if successful
+        members = group.get_pennkey_active_members()
+        if len(members) < 1:
+            return {
+                "complete_success": False,
+                "partial_success": False,
+                "error": "No members in the group have enabled their pennkey to be used for group bookings",
+            }
+
+        # randomize order, and put requester first in the order
+        random.shuffle(members)
+        for (i, member) in enumerate(members):  # puts the requester as the first person
+            if member["username"] == requester_pennkey:
+                temp = members[0]
+                members[0] = member
+                members[i] = temp
+                break
+
+        if is_wharton:  # huntsman reservation
+            return {
+                "complete_success": False,
+                "partial_success": False,
+                "error": "Unable to book huntsman rooms yet",
+            }
+        else:  # lib reservation
+            # Find the first timeslot to book for (next_start, next_end)
```

Comment:
eventually will be nice to pull out `book_wharton` and `book_libcal` into separate functions for readability.

Reply:
Agreed - although I have a feeling they are going to share a decent amount of code